extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -516,58 +481,46 @@ export class AmpStoryReaction extends AMP.BaseElement {
    *    ...
    *  ]
    * }
-   * @param {ReactionResponseType|undefined} response
+   * @param {ReactionResponseType} response
    * @private
    */
   handleSuccessfulDataRetrieval_(response) {
-    if (!(response && 'data' in response)) {
+    if (!(response && 'options' in response)) {
       dev().error(
         TAG,
-        `Invalid reaction response, expected { data: ReactionResponseType, ...} but received ${response}`
+        `Invalid reaction response, expected { data: Array<ReactionOptionType>, ...} but received ${response}`
       );
       return;
-    }
-
-    this.responseData_ = response.data;
-
-    this.hasUserSelection_ = this.responseData_.hasUserResponded;
-    if (this.hasUserSelection_) {
-      this.updateReactionOnDataRetrieval_(response.data);
+    } else {
+      this.updateComponentOnDataRetrieval_(
+        response['options'].splice(0, this.optionElements_.length)
```

Comment:
I think having the "The user-selected reaction could not be found" error message was much better for developers. Now, if their backend returns bad data it'll just fail silently.

Reply:
I'm not sure what error this refers to, it still fails if the response doesn't have the list of options. Are you talking about the error where the length of the options from the backend is different from the length of options in the frontend? In that case, I assumed it was fine since the backend could always return 4 options even if the quiz had only two options, and it would ignore the last two.